lib/tokenize.cpp

```diff
@@ -4708,6 +4708,10 @@ void Tokenizer::setVarIdPass1()
 
             if (!isC() && Token::simpleMatch(tok2, "const new"))
                 continue;
+            if (!isC() && Token::simpleMatch(tok, "static_assert (")) {
```

Comment:
I didn't look carefully but this feels rather hacky to me. So we would still get such assertions for:
```
something(true && false);
```
spontanously `true` should not be seen as a valid variable type and `false` should not be seen as a valid variable name if it's a c++ file.. but I assume you looked into that already?

Reply:
This `something(true && false);` seems to work as expected in an executable scope, but of course `static_assert` can appear in global or other scopes.

Reply:
C23+ also has `static_assert`.

Reply:
And you can achieve a static assertion with a macro. I have seen such:

    #define STATIC_ASSERT(X)    char  dummy  [ (X) ? 1 : -1 ]

Reply:
My comment about the global scope was intended as a justification why `static_assert` needs special handling.

Reply:
my first question was if you looked into if `true` and `false` can be handled more intelligently instead? It does not seem impossible to determine that `true && false` is not a C++ variable declaration.

Reply:
> my first question was if you looked into if `true` and `false` can be handled more intelligently instead? It does not seem impossible to determine that `true && false` is not a valid C++ variable declaration.

This is, among other things, tied to `%type%` matching `true|false`. I didn't want to touch that in this PR.

Reply:
> yeah .. I wanted to point out it would also be good to cover STATIC_ASSERT etc..
> 
> so such testcase would be good imho:

I believe this currently leads to an internalError. But there were no such errors in daca when I last checked, so code like that seems to be uncommon in C++.